databricks_cli/configure/config.py

```diff
@@ -28,54 +28,39 @@
 
 from databricks_cli.configure.provider import DEFAULT_SECTION, get_config_for_profile
 from databricks_cli.utils import error_and_quit
-from databricks_cli.sdk import ApiClient, DbfsService, WorkspaceService, JobsService, \
-    ClusterService, ManagedLibraryService
+from databricks_cli.sdk import ApiClient
 
 
 def require_config(function):
+    """
+    All callbacks wrapped by require_config expect the argument ``profile`` to be passed in.
+    """
     @six.wraps(function)
     def decorator(*args, **kwargs):
-        config = get_config_for_profile(DEFAULT_SECTION)
+        profile = kwargs.pop('profile')
+        config = get_config_for_profile(profile)
         if not config.is_valid:
-            error_and_quit(('You haven\'t configured the CLI yet! '
-                            'Please configure by entering `{} configure`').format(sys.argv[0]))
+            if profile == DEFAULT_SECTION:
+                error_and_quit(('You haven\'t configured the CLI yet! ' +
+                                'Please configure by entering `{} configure`').format(sys.argv[0]))
+            else:
+                error_and_quit(('You haven\'t configured the CLI yet for the ' +
+                                'profile {}! '.format(profile) +
+                                'Please configure by entering `{} configure --profile {}`').format(sys.argv[0], profile)) # noqa
+        kwargs['api_client'] = _get_api_client(config)
+
         return function(*args, **kwargs)
     decorator.__doc__ = function.__doc__
     return decorator
 
 
 def profile_option(f):
-    return click.option('--profile', required=False, default=DEFAULT_SECTION)(f)
+    return click.option('--profile', required=False, default=DEFAULT_SECTION,
+                        help='Connection configuration to use.')(f)
```

Comment:
CLI connection profile to use. See `databricks configure -h`. Should also mention that if this is case sensitive or not.